fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -240,6 +249,24 @@ private void insertField(DocumentEntry entry, final Document document) {
         }
     }
 
+    public <M extends Message> String extractGrouping(@Nullable FDBIndexableRecord<M> record) {
+        KeyExpression expression = state.index.getRootExpression();
+        if (expression instanceof GroupingKeyExpression) {
+            final List<DocumentEntry> groupingExpressions = getFields(((GroupingKeyExpression)expression).getGroupingSubKey(), record, record.getRecord());
+            if (groupingExpressions.isEmpty()) {
+                return null;
+            }
+            String indexPreface = "";
+            Key.Evaluated evaluatedKey = ((GroupingKeyExpression)expression).getGroupingSubKey().evaluateSingleton(record);
+            if (!evaluatedKey.containsNonUniqueNull()) {
+                Object value = evaluatedKey.values().get(0);
```

Comment:
This doesn't appear to match `scan`, which takes all the keys. If I'm right, then a test with a multi-field grouping should be failing. If we don't have one like that, perhaps we should.